static/app/views/starfish/views/spanSummary/sidebar.tsx

```diff
@@ -0,0 +1,235 @@
+import {useTheme} from '@emotion/react';
+import styled from '@emotion/styled';
+import {useQuery} from '@tanstack/react-query';
+import moment from 'moment';
+
+import Duration from 'sentry/components/duration';
+import {t} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import {formatPercentage} from 'sentry/utils/formatters';
+import Chart from 'sentry/views/starfish/components/chart';
+import {HOST} from 'sentry/views/starfish/modules/APIModule/APIModuleView';
+import {
+  getEndpointDetailSeriesQuery,
+  getEndpointDetailTableQuery,
+} from 'sentry/views/starfish/modules/APIModule/queries';
+import {zeroFillSeries} from 'sentry/views/starfish/utils/zeroFillSeries';
+import {getUniqueTransactionCountQuery} from 'sentry/views/starfish/views/spanSummary/queries';
+
+export default function Sidebar({description, transactionName}) {
+  const theme = useTheme();
+  const seriesQuery = getEndpointDetailSeriesQuery({
+    description,
+    transactionName,
+  });
+  const aggregatesQuery = getEndpointDetailTableQuery({
+    description,
+    transactionName,
+  });
+
+  // This is supposed to a metrics span query that fetches aggregate metric data
+  const {isLoading: _isLoadingSideBarAggregateData, data: aggregateData} = useQuery({
+    queryKey: [aggregatesQuery],
+    queryFn: () => fetch(`${HOST}/?query=${aggregatesQuery}`).then(res => res.json()),
+    retry: false,
+    initialData: [],
+  });
+
+  // Also a metrics span query that fetches series data
+  const {isLoading: isLoadingSeriesData, data: seriesData} = useQuery({
+    queryKey: [seriesQuery],
+    queryFn: () => fetch(`${HOST}/?query=${seriesQuery}`).then(res => res.json()),
+    retry: false,
+    initialData: [],
+  });
+
+  // This is a metrics request on transactions data!
+  // We're fetching the count of events on a specific transaction so we can
+  // calculate span frequency using metrics spans vs metrics transactions
```

Comment:
Does this need to take the span hash into account? As in, should it be fetching only transactions that include the span?

Reply:
This doesn't use the span hash, only the transaction name! This is because I purposely want to fetch all events belonging to this transaction, including events that don't have any instance of this span. This is so I can calculate the span frequency ie `count_unique of span / total event count`. If I only fetched the transactions that include the span, the span frequency would always be 100%